pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -303,35 +303,44 @@ func (h *Handler) createNad(setting *harvesterv1.Setting) (string, error) {
 	// create nad
 	var nadResult *nadv1.NetworkAttachmentDefinition
 	if nadResult, err = h.networkAttachmentDefinitions.Create(&nad); err != nil {
-		return "", fmt.Errorf("create net-attach-def failed %v", err)
+		return "", "", fmt.Errorf("create net-attach-def failed %v", err)
 	}
 
-	return fmt.Sprintf("%s/%s", nadResult.Namespace, nadResult.Name), nil
+	return nadResult.Namespace, nadResult.Name, nil
 }
 
 func (h *Handler) checkValueIsChanged(setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
 	var updatedSetting *harvesterv1.Setting
 	var err error
-	nadName := ""
+	var nadNamespace, nadName string
+	nad := ""
 
 	if h.checkIsSameHashValue(setting) {
 		return setting, nil
 	}
 
 	if setting.Value != "" {
-		nadName, err = h.createNad(setting)
+		nadNamespace, nadName, err = h.createNad(setting)
```

Comment:
Can we check the existence of the NAD and skip creating a new one, instead of deleting the NAD?

Reply:
It will lose track because the controller may fail to record the new NAD in the storage network setting annotation. 

IMHO, the safest way is to delete the NAD which can't be correctly recorded. Or do you have a better opinion? @bk201 

Thanks.

Reply:
Maybe we don't have to track the NAD for the storage-network setting because there will be only one NAD for the storage network. Is it possible to add a label to denote such a created NAD is for the storage network? With that, it should be possible to list NADs with a label filter. Just my two cents :P

Reply:
Good idea, I add the NAD's label with the hash from the setting value, and check if the NAD exists with the required hash in each reconcile, thanks.